lib/Conversion/LLHDToLLVM/LLHDToLLVM.cpp

```diff
@@ -259,12 +273,12 @@ static void insertPersistence(LLVMTypeConverter &converter,
       auto gep0 = rewriter.create<LLVM::GEPOp>(
           loc, elemTy.getPointerTo(), converted.getArgument(1),
           ArrayRef<Value>({zeroC0, threeC0, indC0}));
-      rewriter.create<LLVM::StoreOp>(loc, op->getResult(0), gep0);
+      rewriter.create<LLVM::StoreOp>(loc, toStore, gep0);
 
       // Load the value from the persistence table and substitute the original
-      // use with it.
-      for (auto &use : op->getUses()) {
-        if (dom.properlyDominates(op->getBlock(), use.getOwner()->getBlock())) {
+      // use with it, whenever it is in a different block.
+      for (auto &use : llvm::make_early_inc_range(op->getUses())) {
+        if (op->getBlock() != use.getOwner()->getBlock()) {
```

Comment:
Looks like a reasonably pessimistic condition to "spill over" into the persisted space. Would we be able to use a more accurate heuristic down the line, like checking if the *temporal regions* differ?

Reply:
Yes, this is probably the most conservative approach possible. Using temporal region analysis would already decrease the redundancy in the persisted values considerably I think. Also looking into coroutines would be great further down the line, and maybe would even allow us to get rid of this part of the lowering altogether!

Reply:
Anyway I think the current approach is very valuable to have as a first shot that works!